process explorer/backend/LocalCollector/InfoAggregator.cs

```diff
@@ -0,0 +1,69 @@
+﻿/* Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. */
+
+using ProcessExplorer.Entities;
+using ProcessExplorer.Entities.Connections;
+using ProcessExplorer.Entities.EnvironmentVariables;
+using ProcessExplorer.Entities.Modules;
+using ProcessExplorer.Entities.Registrations;
+using System.Net.Http.Headers;
+
+namespace ProcessExplorer
+{
+    public class InfoAggregator
+    {
+        InfoAggregator()
+        {
+            Data = new InfoAggregatorDto();
+        }
+        public InfoAggregator(Guid id, AppUserInfoDto user, EnvironmentMonitorDto envs, ConnectionMonitor cons)
+            :this()
+        {
+            Data.Id = id;
+            Data.User = user;
+            Data.EnvironmentVariables = envs;
+            Data.Connections = cons.Data;
+        }
+        public InfoAggregator(Guid id, AppUserInfoDto user, EnvironmentMonitorDto envs, ConnectionMonitor cons,
+            RegistrationMonitorDto registrations, ModuleMonitorDto modules)
+            : this(id, user, envs, cons)
+        {
+            Data.Registrations = registrations;
+            Data.Modules = modules;
+        }
+        
+        public InfoAggregatorDto Data { get; set; }
+
+        //SAMPLE MESSAGE SENDING
+        private async Task SendMessage(string url)
+        {
+            using (var client = new HttpClient())
```

Comment:
I'm not sure I understand the context here but it seems you're using HttpClient wrong: see https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests for details